### PR TITLE
Allowed $0.00 price bids if there are deals

### PR DESCRIPTION
**Pull request by AlexBVolcy**
Addressing this issue https://github.com/prebid/prebid-server/issues/1308

### CODE REVIEW

File: exchange/bidder_validate_bids.go [hunk 1]
```diff
@@ -110,8 +110,11 @@ func validateBid(bid *pbsOrtbBid) (bool, error) {
 	if bid.bid.ImpID == "" {
 		return false, fmt.Errorf("Bid \"%s\" missing required field 'impid'", bid.bid.ID)
 	}
-	if bid.bid.Price <= 0.0 {
-		return false, fmt.Errorf("Bid \"%s\" does not contain a positive 'price'", bid.bid.ID)
+	if bid.bid.Price < 0.0 {
+		return false, fmt.Errorf("Bid \"%s\" does not contain a positive (or zero if there is a deal) 'price'", bid.bid.ID)
+	}
+	if bid.bid.Price <= 0.0 && bid.bid.DealID == "" {
```
Comment (hhhjort) [inline exchange/bidder_validate_bids.go:116]: You should be able to have 'bid.bid.Price == 0.0 && bid.bid.DealID', I don't think it is possible for both 'x < 0.0' and 'x == 0.0' to be false yet 'x <= 0.0' be true.

**Comment by mansinahar** [comment]
The issue suggests adding a `allowzerovaluebids` flag that skips this check for $0.00 bids for deals. However, I don't see this implemented as part of the Java PR for this either but can you please check with @bretg to confirm that's not needed?

File: exchange/bidder_validate_bids.go (hunk 1 above)
Comment (AlexBVolcy) [reply exchange/bidder_validate_bids.go:116]: Ah yeah that makes sense. Will update!

Comment (AlexBVolcy) [reply exchange/bidder_validate_bids.go]: Just updated!

**Review by hhhjort — APPROVED** [review]
LGTM assuming Mansi's concern is cleared.

**Review by mansinahar — APPROVED** [review]

**Pull request merged**